core/src/test/java/io/cucumber/core/options/CucumberOptionsAnnotationParserTest.java

```diff
@@ -218,6 +220,44 @@ void create_with_glue() {
             contains(uri("classpath:/app/features/user/registration"), uri("classpath:/app/features/hooks")));
     }
 
+    @Test
+    void warn_when_features_as_filesystem_path_to_resources() {
+        // warn when 'src/{test,main}/resources' is used
+
+        LogRecordListener logRecordListener = new LogRecordListener();
+        LoggerFactory.addListener(logRecordListener);
+
+        FeaturePath.parse("src/test/resources/features");
+        FeaturePath.parse("src/main/resources/features/feature1");
+
+        LoggerFactory.removeListener(logRecordListener);
+
+        assertAll(
+            () -> assertTrue(LogRecordListener.anyRecordMessageMatch(logRecordListener,
+                ".*replace.*src/test/resources/features.*'classpath:features'.*")),
+            () -> assertTrue(LogRecordListener.anyRecordMessageMatch(logRecordListener,
+                ".*replace.*src/main/resources/features/feature1.*'classpath:features/feature1'.*")));
+    }
+
+    @Test
+    void warn_when_glue_as_filesystem_path() {
+        // warn when 'src/{test,main}/{java,kotlin,scala,groovy}' is used
+
+        LogRecordListener logRecordListener = new LogRecordListener();
+        LoggerFactory.addListener(logRecordListener);
+
+        GluePath.parse("src/main/java/com/example");
+        GluePath.parse("src/test/java/com/package/other_package");
```

Comment:
Because we're testing multiple inputs for a function of `path -> log message` consider using [Parameterized tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests), specifically the `testWithMultiArgMethodSource` example.

This makes it possible to test multiple different inputs/outputs for a function while also having a single test for each case.